--- smartmontools-6.5/drivedb.h.orig	2016-05-07 17:33:02.820674826 -0500
+++ smartmontools-6.5/drivedb.h	2016-05-07 17:46:54.208759606 -0500
@@ -857,6 +857,21 @@
     "-v 245,raw48,Max_Erase_Count "
     "-v 246,raw48,Total_Erase_Count "
   },
+  { "Phison S9 SSD", // see MKP_521_Phison_SMART_attribute.pdf
+    ".*SATA SSD", // PC Engines msata16d, MyDigitalSSD Super-Boot-Drive mSATA
+    "S9FM.*", "",
+  //"-v 1,raw48,Raw_Read_Error_Rate "
+  //"-v 9,raw24(raw8),Power_On_Hours "
+  //"-v 12,raw48,Power_Cycle_Count "
+    "-v 168,raw48,SATA_Phy_Error_Count "
+    "-v 170,raw24/raw24:z54z10,Bad_Blk_Ct_Erl/Lat " // Early bad block/Later bad block
+    "-v 173,raw16(avg16),MaxAvgErase_Ct "
+    "-v 192,raw48,Unsafe_Shutdown_Count "
+  //"-v 194,tempminmax,Temperature_Celsius "
+    "-v 196,raw48,Not_In_Use "
+    "-v 218,raw48,CRC_Error_Count "
+    "-v 241,raw48,Lifetime_Writes_MiB "
+  },
   { "Indilinx Barefoot based SSDs",
     "Corsair CSSD-V(32|60|64|128|256)GB2|" // Corsair Nova, tested with Corsair CSSD-V32GB2/2.2
     "Corsair CMFSSD-(32|64|128|256)D1|" // Corsair Extreme, tested with Corsair CMFSSD-128D1/1.0
