--- linux-custom/kernel/rcutree.c.orig	2016-05-17 11:18:36.318141162 -0500
+++ linux-custom/kernel/rcutree.c	2016-05-17 11:21:36.375884207 -0500
@@ -210,7 +210,7 @@
 module_param(qhimark, long, 0);
 module_param(qlowmark, long, 0);
 
-int rcu_cpu_stall_suppress __read_mostly;
+int rcu_cpu_stall_suppress __read_mostly = 1; /* 1 = suppress stall warnings. */
 module_param(rcu_cpu_stall_suppress, int, 0644);
 
 static void force_quiescent_state(struct rcu_state *rsp, int relaxed);
