--- shellinabox-239/shellinabox/vt100.js.orig	2011-11-30 22:55:50.000000000 -0600
+++ shellinabox-239/shellinabox/vt100.js	2011-11-30 22:59:20.000000000 -0600
@@ -289,6 +289,10 @@
     this.signature          = Math.floor(16807*this.signature + 1) %
                                          ((1 << 31) - 1);
   }
+  var isMobile = navigator.userAgent.match(/iPad|iPhone|iPod/i) != null;
+  if (isMobile) {
+    this.softKeyboard = true;
+  }
   if (typeof userCSSList != 'undefined') {
     for (var i = 0; i < userCSSList.length; ++i) {
       var label             = userCSSList[i][0];
@@ -971,6 +971,10 @@
                                   document.documentElement.clientWidth ||
                                   document.body.clientWidth) -
                                  marginRight != x + this.container.offsetWidth;
+  var isMobile = navigator.userAgent.match(/iPad|iPhone|iPod/i) != null;
+  if (isMobile) {
+    this.isEmbedded = false;
+  }
   if (!this.isEmbedded) {
     // Some browsers generate resize events when the terminal is first
     // shown. Disable showing the size indicator until a little bit after
