--- igb-5.2.9.2/src/igb_main.c.orig	2014-07-29 17:53:55.000000000 -0500
+++ igb-5.2.9.2/src/igb_main.c	2014-07-29 17:55:09.000000000 -0500
@@ -2543,7 +2543,7 @@
 	u8 pba_str[E1000_PBANUM_LENGTH];
 	s32 ret_val;
 	static int global_quad_port_a; /* global quad port a indication */
-	int i, err, pci_using_dac;
+	int err, pci_using_dac;
 	static int cards_found;
 
 	err = pci_enable_device_mem(pdev);
@@ -2929,10 +2929,9 @@
 		  (hw->bus.width == e1000_bus_width_pcie_x2) ? "Width x2" :
 		  (hw->bus.width == e1000_bus_width_pcie_x1) ? "Width x1" :
 		  (hw->mac.type == e1000_i354) ? "integrated" : "unknown"));
-	dev_info(pci_dev_to_dev(pdev), "%s: MAC: ", netdev->name);
-	for (i = 0; i < 6; i++)
-		netdev_info(netdev, "%2.2x%c", netdev->dev_addr[i], i ==
-			    5 ? '\n' : ':');
+	netdev_info(netdev, "MAC: %2.2x:%2.2x:%2.2x:%2.2x:%2.2x:%2.2x\n",
+		netdev->dev_addr[0], netdev->dev_addr[1], netdev->dev_addr[2],
+		netdev->dev_addr[3], netdev->dev_addr[4], netdev->dev_addr[5]);
 
 	ret_val = e1000_read_pba_string(hw, pba_str, E1000_PBANUM_LENGTH);
 	if (ret_val)
