--- igb-5.7.2/src/kcompat_impl.h.orig	2021-08-23 13:05:17.883225072 -0500
+++ igb-5.7.2/src/kcompat_impl.h	2021-08-23 13:07:46.818934814 -0500
@@ -43,10 +43,12 @@
  * old implementation of skb_frag_t is trivial.
  */
 #ifdef NEED_SKB_FRAG_OFF_ACCESSORS
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(4,19,201))
 static inline unsigned int skb_frag_off(const skb_frag_t *frag)
 {
 	return frag->page_offset;
 }
+#endif /* 4.19.201 */
 
 static inline void skb_frag_off_add(skb_frag_t *frag, int delta)
 {
