--- libusb-1.0.26/configure.ac.orig	2022-04-23 13:39:10.807739729 -0500
+++ libusb-1.0.26/configure.ac	2022-04-23 13:40:05.248820965 -0500
@@ -153,7 +153,6 @@
 	AC_SEARCH_LIBS([pthread_create], [pthread],
 		[test "x$ac_cv_search_pthread_create" != "xnone required" && AC_SUBST(THREAD_LIBS, [-lpthread])],
 		[], [])
-	AC_SEARCH_LIBS([__atomic_fetch_add_4], [atomic])
 elif test "x$platform" = xwindows; then
 	AC_DEFINE([PLATFORM_WINDOWS], [1], [Define to 1 if compiling for a Windows platform.])
 else
