--- monit-5.25.2/configure.ac.orig	2018-12-18 10:16:49.636344222 -0600
+++ monit-5.25.2/configure.ac	2018-12-18 10:21:10.916071261 -0600
@@ -479,14 +479,7 @@
 
 # Find the right directory to put the root-mode PID file in
 AC_MSG_CHECKING([pid file location])
-if test -d "/run"
-then
-	piddir="/run"
-elif test -d "/var/run"; then
-        piddir="/var/run"
-elif test -d "/etc"; then
-        piddir="/etc"
-fi
+piddir="/var/run"
 
 AC_DEFINE_UNQUOTED([PIDDIR], "$piddir",
 	  [Define to the pid storage directory.])
