--- monit-5.11/libmonit/src/system/NetStatistics.c.orig	2014-12-24 13:59:56.000000000 -0600
+++ monit-5.11/libmonit/src/system/NetStatistics.c	2014-12-24 14:39:16.000000000 -0600
@@ -71,6 +71,8 @@
 
 #define T NetStatistics_T
 
+#define DO_FIX_32BIT 1
+#define NO_FIX_32BIT 0
 
 static struct {
         struct ifaddrs *addrs;
@@ -81,6 +83,7 @@
 typedef struct NetStatisticsData_T {
         long long last;
         long long now;
+        long long raw;
         long long minute[60];
         long long hour[24];
 } NetStatisticsData_T;
@@ -122,6 +125,22 @@
 /* --------------------------------------------------------------- Private */
 
 
+static void _updateRawStats(NetStatisticsData_T *data, long long value, int fix32bit) {
+	long long raw = value;
+
+	if (fix32bit == DO_FIX_32BIT) {
+		if (raw < data->raw) {
+			raw += 4294967296LL;
+		}
+	}
+	data->last = data->now;
+	if (raw > data->raw) {
+		data->now += raw - data->raw;
+	}
+	data->raw = value;
+}
+
+
 #if defined DARWIN
 #include "os/macosx/NetStatistics.c"
 #elif defined FREEBSD
@@ -153,6 +172,18 @@
         S->state = -1;
         S->duplex = -1;
         S->speed = 0LL;
+        S->ibytes.now = 0LL;
+        S->ipackets.now = 0LL;
+        S->ierrors.now = 0LL;
+        S->obytes.now = 0LL;
+        S->opackets.now = 0LL;
+        S->oerrors.now = 0LL;
+        S->ibytes.raw = 0LL;
+        S->ipackets.raw = 0LL;
+        S->ierrors.raw = 0LL;
+        S->obytes.raw = 0LL;
+        S->opackets.raw = 0LL;
+        S->oerrors.raw = 0LL;
         _resetData(&(S->ibytes), -1LL);
         _resetData(&(S->ipackets), -1LL);
         _resetData(&(S->ierrors), -1LL);
--- monit-5.11/libmonit/src/system/os/linux/NetStatistics.c.orig	2014-12-24 14:00:18.000000000 -0600
+++ monit-5.11/libmonit/src/system/os/linux/NetStatistics.c	2014-12-24 14:55:04.000000000 -0600
@@ -90,18 +90,12 @@
                         if (sscanf(buf, "%256[^:]: %lld %lld %lld %*s %*s %*s %*s %*s %lld %lld %lld %*s %*s %*s %*s %*s", iface, &ibytes, &ipackets, &ierrors, &obytes, &opackets, &oerrors) == 7 && Str_isEqual(Str_trim(iface), interface)) {
                                 S->timestamp.last = S->timestamp.now;
                                 S->timestamp.now = Time_milli();
-                                S->ipackets.last = S->ipackets.now;
-                                S->ipackets.now = ipackets;
-                                S->ibytes.last = S->ibytes.now;
-                                S->ibytes.now = ibytes;
-                                S->ierrors.last = S->ierrors.now;
-                                S->ierrors.now = ierrors;
-                                S->opackets.last = S->opackets.now;
-                                S->opackets.now = opackets;
-                                S->obytes.last = S->obytes.now;
-                                S->obytes.now = obytes;
-                                S->oerrors.last = S->oerrors.now;
-                                S->oerrors.now = oerrors;
+				_updateRawStats(&(S->ibytes), ibytes, DO_FIX_32BIT);
+				_updateRawStats(&(S->ipackets), ipackets, DO_FIX_32BIT);
+				_updateRawStats(&(S->ierrors), ierrors, DO_FIX_32BIT);
+				_updateRawStats(&(S->obytes), obytes, DO_FIX_32BIT);
+				_updateRawStats(&(S->opackets), opackets, DO_FIX_32BIT);
+				_updateRawStats(&(S->oerrors), oerrors, DO_FIX_32BIT);
                                 fclose(f);
                                 return true;
                         }
