From 18223cd0949d582628aeeced50902424b7d44a41 Mon Sep 17 00:00:00 2001
From: Quentin Armitage <quentin@armitage.org.uk>
Date: Fri, 19 Mar 2021 23:19:56 +0000
Subject: [PATCH] configure: Fix uninitialised WITH_TRACK_PROCESS in
 configure.ac

Signed-off-by: Quentin Armitage <quentin@armitage.org.uk>
---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index 4d3b1eca3..0c4d7764f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1859,6 +1859,7 @@ MACVLAN_SUPPORT=No
 ENABLE_JSON=No
 BFD_SUPPORT=No
 HAVE_CN_PROC=No
+WITH_TRACK_PROCESS=No
 if test "$enable_vrrp" != no; then
   VRRP_SUPPORT=Yes
   AC_DEFINE([_WITH_VRRP_], [ 1 ], [Define to 1 if have VRRP support])
