From 96cf5aa30c2a11f2206240278ecff45fcd31d88c Mon Sep 17 00:00:00 2001
From: Quentin Armitage <quentin@armitage.org.uk>
Date: Mon, 25 Mar 2019 14:29:24 +0000
Subject: [PATCH] Fix rpmbuild on CentOS7, and rely on auto-requires

---
 configure.ac       | 25 +++++++++++++++++++++++++
 keepalived.spec.in | 20 ++++++++++++--------
 2 files changed, 37 insertions(+), 8 deletions(-)

diff --git a/configure.ac b/configure.ac
index 0c098597a..8bf06a8e1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -762,6 +762,27 @@ AC_LINK_IFELSE([AC_LANG_SOURCE([[
     LDFLAGS=$SAV_LDFLAGS
   ])
 
+# Check if fpclassify() requires -lm
+CFLAGS=
+LDFLAGS=
+AC_MSG_CHECKING([whether fpclassify() requires -lm])
+AC_LINK_IFELSE([AC_LANG_SOURCE([[
+  # include <math.h>
+  int main(int argc, char **argv)
+  {
+    double zero = 0.0;
+    if (fpclassify(zero) == FP_ZERO)
+      return 0;
+    return 1;
+  }
+  ]])],
+  AC_MSG_RESULT([no]),
+  AC_MSG_RESULT([yes])
+    add_to_var([KA_LDFLAGS], [-lm])
+  )
+CFLAGS=$SAV_CFLAGS
+LDFLAGS=$SAV_LDFLAGS
+
 # Checks for header files.
 AC_CHECK_HEADERS([arpa/inet.h fcntl.h limits.h netdb.h netinet/in.h stdint.h stdlib.h string.h sys/ioctl.h sys/param.h sys/prctl.h sys/socket.h sys/time.h syslog.h unistd.h],
   [], [AC_MSG_ERROR([Missing/unusable system header file <$ac_header>])])
@@ -1489,6 +1510,10 @@ AM_CONDITIONAL([LIBIPTC], [test $USE_LIBIPTC = Yes])
 AM_CONDITIONAL([LIBIPSET], [test $USE_LIBIPSET = Yes])
 AM_CONDITIONAL([IPTABLES], [test $USE_IPTABLES = Yes])
 AM_CONDITIONAL([IPTABLES_CMD], [test $USE_IPTABLES = Yes -a \( $USE_LIBIPTC = No -o .$LIBIPTC_DYNAMIC = .Yes \)])
+AM_CONDITIONAL([LIBIPTC_DYNAMIC], [test $USE_LIBIPTC = Yes -a .$LIBIPTC_DYNAMIC = .Yes])
+AM_CONDITIONAL([LIBIPSET_DYNAMIC], [test $USE_LIBIPSET = Yes -a .${enable_libipset_dynamic} != .no])
+AM_CONDITIONAL([LIBXTABLES_DYNAMIC], [test $USE_LIBIPTC = Yes -a .$enable_libxtables_dynamic = .yes])
+AM_CONDITIONAL([REQUIRE_IPTABLES_LIBS], [test $USE_IPTABLES = Yes -a $USE_LIBIPTC = Yes -a \( .$LIBIPTC_DYNAMIC = .Yes -o .$enable_libxtables_dynamic = .yes \)])
 unset LIBS
 
 dnl ----[Check for nftables libraries]----
diff --git a/keepalived.spec.in b/keepalived.spec.in
index 48f2a995d..656ad7899 100644
--- a/keepalived.spec.in
+++ b/keepalived.spec.in
@@ -29,14 +29,14 @@ BuildRequires: autoconf automake pkgconfig
 @INIT_SYSTEMD_TRUE@BuildRequires: systemd-units
 @WITH_REGEX_TRUE@BuildRequires: pcre2-devel
 
-@SNMP_TRUE@Requires: net-snmp
-@WITH_DBUS_TRUE@Requires: glib2
-@LIBIPTC_TRUE@Requires: iptables-libs
-@LIBIPSET_TRUE@Requires: ipset-libs
-@LIBNL1_TRUE@Requires: libnl
-@LIBNL3_TRUE@Requires: libnl3
-@NFTABLES_TRUE@Requires: libnftnl libmnl
-@WITH_REGEX_TRUE@Requires: pcre2
+@REQUIRE_IPTABLES_LIBS_TRUE@%if 0%{?rhel} <= 7
+@REQUIRE_IPTABLES_LIBS_TRUE@Requires: iptables
+@REQUIRE_IPTABLES_LIBS_TRUE@%else
+@REQUIRE_IPTABLES_LIBS_TRUE@Requires: iptables-libs
+@REQUIRE_IPTABLES_LIBS_TRUE@%endif
+@LIBIPSET_DYNAMIC_TRUE@Requires: ipset-libs
+@LIBNL_DYNAMIC_TRUE@@LIBNL1_TRUE@Require: libnl
+@LIBNL_DYNAMIC_TRUE@@LIBNL3_TRUE@Require: libnl3
 
 %description
 The main goal of the keepalived project is to add a strong & robust keepalive
@@ -65,6 +65,10 @@ CONFIG_OPTS=
 @IPTABLES_FALSE@CONFIG_OPTS="$CONFIG_OPTS --disable-iptables"
 @IPTABLES_TRUE@@LIBIPTC_FALSE@CONFIG_OPTS="$CONFIG_OPTS --disable-libiptc"
 @IPTABLES_TRUE@@LIBIPSET_FALSE@CONFIG_OPTS="$CONFIG_OPTS --disable-libipset"
+@LIBIPTC_DYNAMIC_TRUE@CONFIG_OPTS="$CONFIG_OPTS --enable-libiptc-dynamic"
+@LIBXTABLES_DYNAMIC_TRUE@CONFIG_OPTS="$CONFIG_OPTS --enable-libxtables-dynamic"
+@LIBIPSET_DYNAMIC_FALSE@CONFIG_OPTS="$CONFIG_OPTS --disable-libipset-dynamic"
+@LIBNL_DYNAMIC_TRUE@CONFIG_OPTS="$CONFIG_OPTS --enable-libnl-dynamic"
 @SNMP_KEEPALIVED_TRUE@CONFIG_OPTS="$CONFIG_OPTS --enable-snmp"
 @SNMP_KEEPALIVED_FALSE@@SNMP_VRRP_TRUE@CONFIG_OPTS="$CONFIG_OPTS --enable-snmp-vrrp"
 @SNMP_KEEPALIVED_FALSE@@SNMP_CHECKER_TRUE@CONFIG_OPTS="$CONFIG_OPTS --enable-snmp-checker"
