--- fping-4.1/src/fping.c.orig	2018-10-20 15:32:59.713335391 -0500
+++ fping-4.1/src/fping.c	2018-10-20 15:52:14.331484240 -0500
@@ -362,6 +362,11 @@
     socket4 = open_ping_socket_ipv4(ping_data_size);
 #ifdef IPV6
     socket6 = open_ping_socket_ipv6(ping_data_size);
+    /* if called (sym-linked) via 'fping6', imply '-6'
+     * for backward compatibility */
+    if (strstr(prog, "fping6")) {
+        hints_ai_family = AF_INET6;
+    }
 #endif
 
     if ((uid = getuid())) {
