--- fping-4.1/src/fping.c.orig	2018-10-21 07:48:47.996503897 -0500
+++ fping-4.1/src/fping.c	2018-10-21 07:54:53.582797143 -0500
@@ -692,7 +692,11 @@
 
     /* validate various option settings */
 
-    if (socket4 < 0 && socket6 < 0) {
+#ifndef IPV6
+    if (socket4 < 0) {
+#else
+    if ((socket4 < 0 && socket6 < 0) || (hints_ai_family == AF_INET6 && socket6 < 0)) {
+#endif
         crash_and_burn("can't create socket (must run as root?)");
     }
 
