--- fossil-2.22/src/cgi.c.orig	2023-06-02 13:07:36.184673568 -0500
+++ fossil-2.22/src/cgi.c	2023-06-02 13:10:45.912932271 -0500
@@ -1995,6 +1995,12 @@
   if( zToken==0 ){
     malformed_request("malformed URL in HTTP header");
   }
+  const char *proxy_base = "/admin/fossil/"; /* str must begin and end with '/' */
+  if( (i = strlen(proxy_base))>1 ){
+    if( strncmp(zToken, proxy_base, i)==0 ){
+      memmove(zToken, zToken+i-1, strlen(zToken)-i+2);
+    }
+  }
   cgi_setenv("REQUEST_URI", zToken);
   cgi_setenv("SCRIPT_NAME", "");
   for(i=0; zToken[i] && zToken[i]!='?'; i++){}
--- fossil-2.22/src/main.c.orig	2023-06-02 13:11:11.306483891 -0500
+++ fossil-2.22/src/main.c	2023-06-02 13:16:02.106112709 -0500
@@ -1417,8 +1417,8 @@
     zCur = PD("SCRIPT_NAME","/");
     i = strlen(zCur);
     while( i>0 && zCur[i-1]=='/' ) i--;
-    if( fossil_stricmp(zMode,"on")==0 ){
-      g.zBaseURL = mprintf("https://%s%.*s", z, i, zCur);
+    if( 1 ){  /* AstLinux */
+      g.zBaseURL = mprintf("https://%s/admin/fossil%.*s", z, i, zCur);
       g.zTop = &g.zBaseURL[8+strlen(z)];
       g.zHttpsURL = g.zBaseURL;
     }else{
--- fossil-2.22/src/login.c.orig	2023-06-02 13:16:15.157346386 -0500
+++ fossil-2.22/src/login.c	2023-06-02 13:19:23.168243120 -0500
@@ -1769,9 +1769,9 @@
     blob_init(&redir, 0, 0);
     if( zPathInfo[0]=='/' ) zPathInfo++; /* skip leading slash */
     if( fossil_wants_https(1) ){
-      blob_appendf(&redir, "%s/login?g=%T", g.zHttpsURL, zPathInfo);
+      blob_appendf(&redir, "%s/login?g=/admin/fossil%T", g.zHttpsURL, zPathInfo);
     }else{
-      blob_appendf(&redir, "%R/login?g=%T", zPathInfo);
+      blob_appendf(&redir, "%R/login?g=/admin/fossil%T", zPathInfo);
     }
     if( zQS && zQS[0] ){
       blob_appendf(&redir, "%%3f%T", zQS);
