--- prosody-0.9.5/Makefile.orig	2014-10-10 10:20:47.000000000 -0500
+++ prosody-0.9.5/Makefile	2014-10-10 10:21:39.000000000 -0500
@@ -37,7 +37,7 @@
 	install -m644 util/*.so $(SOURCE)/util
 	install -d $(SOURCE)/util/sasl
 	install -m644 util/sasl/* $(SOURCE)/util/sasl
-	umask 0022 && cp -r plugins/* $(MODULES)
+	( cd plugins ; find . -type f -exec install -D -m644 '{}' $(MODULES)/'{}' \; )
 	install -m644 certs/* $(CONFIG)/certs
 	install -m644 man/prosodyctl.man $(MAN)/man1/prosodyctl.1
 	test -e $(CONFIG)/prosody.cfg.lua || install -m644 prosody.cfg.lua.install $(CONFIG)/prosody.cfg.lua
