--- iperf3-3.10/configure.ac.orig	2021-05-28 14:05:44.717597927 -0500
+++ iperf3-3.10/configure.ac	2021-05-28 14:06:28.283511949 -0500
@@ -68,7 +68,7 @@
   ISO C90 headers unconditionally.])dnl
 # Autoupdate added the next two lines to ensure that your configure
 # script's behavior did not change.  They are probably safe to remove.
-AC_CHECK_INCLUDES_DEFAULT
+#AC_CHECK_INCLUDES_DEFAULT
 AC_PROG_EGREP
 
 
