--- iprange-2.0.0/configure.ac.orig	2026-04-08 08:41:36.208995849 -0500
+++ iprange-2.0.0/configure.ac	2026-04-08 08:43:08.994178973 -0500
@@ -88,20 +88,6 @@
 	AC_DEFINE_UNQUOTED([unlikely(x)], [(x)], [gcc branch optimization])
 fi
 
-dnl Use -O3 and LTO for GCC
-if test "${GCC}" = "yes"; then
-	CFLAGS=$(echo "${CFLAGS}" | sed 's/-O[[0-9s]]*/-O3/g')
-	case "${CFLAGS}" in
-		*-O*) ;;
-		*) CFLAGS="${CFLAGS} -O3" ;;
-	esac
-	case "${CFLAGS}" in
-		*-flto*) ;;
-		*) CFLAGS="${CFLAGS} -flto"
-		   LDFLAGS="${LDFLAGS} -flto" ;;
-	esac
-fi
-
 if test "${enable_pedantic}" = "yes"; then
 	enable_strict="yes"
 	CFLAGS="${CFLAGS} -pedantic -Wall -Wextra"
