--- stunnel-5.01/configure.ac.orig	2014-04-11 10:43:39.000000000 -0500
+++ stunnel-5.01/configure.ac	2014-04-11 10:49:05.000000000 -0500
@@ -124,7 +124,8 @@
     AC_CHECK_FILE("/dev/ptc", AC_DEFINE([HAVE_DEV_PTS_AND_PTC], [1],
         [Define to 1 if you have '/dev/ptc' device.]))
 else
-    AC_MSG_WARN([cross-compilation: assuming /dev/ptmx and /dev/ptc are not available])
+    AC_DEFINE([HAVE_DEV_PTMX], [1], [Define to 1 if you have '/dev/ptmx' device.])
+    AC_MSG_WARN([cross-compilation: assuming /dev/ptmx is available])
 fi
 
 AC_MSG_NOTICE([**************************************** entropy sources])
@@ -153,7 +154,9 @@
         AC_DEFINE_UNQUOTED([RANDOM_FILE], ["$RANDOM_FILE"], [Random file path])
     fi
 else
-    AC_MSG_WARN([cross-compilation: assuming entropy sources are not available])
+    RANDOM_FILE="/dev/urandom"
+    AC_DEFINE_UNQUOTED([RANDOM_FILE], ["$RANDOM_FILE"], [Random file path])
+    AC_MSG_WARN([cross-compilation: assuming entropy source /dev/urandom is available])
 fi
 
 AC_MSG_NOTICE([**************************************** default group])
