--- miniupnpd-1.6.20120509-orig/genconfig.sh	2012-05-09 06:40:34.000000000 -0400
+++ miniupnpd-1.6.20120509/genconfig.sh	2012-05-24 21:13:04.412605721 -0400
@@ -56,6 +56,12 @@
 	OS_VERSION=$(cat ./os.openwrt)
 fi
 
+# AstLinux special case
+if [ -f ./os.astlinux ]; then
+	OS_NAME=AstLinux
+	OS_VERSION=$(cat ./os.astlinux)
+fi
+
 # Tomato USB special case
 if [ -f ../shared/tomato_version ]; then
 	OS_NAME=Tomato
@@ -223,6 +229,11 @@
 		echo "#define USE_IFACEWATCHER 1" >> ${CONFIGFILE}
 		FW=netfilter
 		;;
+	AstLinux)
+		OS_URL=http://www.astlinux.org/
+		echo "#define USE_IFACEWATCHER 1" >> ${CONFIGFILE}
+		FW=netfilter
+		;;
 	Tomato)
 		OS_NAME=UPnP
 		OS_URL=http://tomatousb.org/
