--- miniupnpd-2.0.20170421/upnpevents.c.orig	2017-07-24 18:31:57.901973795 -0400
+++ miniupnpd-2.0.20170421/upnpevents.c	2017-07-24 18:34:06.948389630 -0400
@@ -473,7 +473,8 @@
 	i = send(obj->s, obj->buffer + obj->sent, obj->tosend - obj->sent, 0);
 	if(i<0) {
 		if(errno != EAGAIN && errno != EWOULDBLOCK && errno != EINTR) {
-			syslog(LOG_NOTICE, "%s: send(): %m", "upnp_event_send");
+			syslog(LOG_NOTICE, "%s: send(%s%s): %m", "upnp_event_send",
+			       obj->addrstr, obj->portstr);
 			obj->state = EError;
 			return;
 		} else {
