--- qemu-guest-agent-2.12.0/qga/commands.c.orig	2018-06-27 09:11:00.203646563 -0500
+++ qemu-guest-agent-2.12.0/qga/commands.c	2018-06-27 09:20:48.079225792 -0500
@@ -51,7 +51,11 @@
 
 void qmp_guest_ping(Error **errp)
 {
-    slog("guest-ping called");
+    static int msgcnt = 8;
+
+    if (msgcnt > 0) {
+        slog("guest-ping called (log muted in %d)", --msgcnt);
+    }
 }
 
 static void qmp_command_info(QmpCommand *cmd, void *opaque)
