--- rng-tools-6.5/rngd.c.orig	2021-04-26 10:04:02.106611557 -0500
+++ rng-tools-6.5/rngd.c	2021-04-26 10:05:39.665189277 -0500
@@ -181,23 +181,23 @@
 	},
 	[JITTER_OPT_BUF_SZ] = {
 		.key = "buffer_size",
-		.int_val = 16535,
+		.int_val = 2501,
 	},
 	[JITTER_OPT_REFILL] = {
 		.key = "refill_thresh",
-		.int_val = 16535,
+		.int_val = 2501,
 	},
 	[JITTER_OPT_RETRY_COUNT] = {
 		.key = "retry_count",
-		.int_val = 1,
+		.int_val = 100,
 	},
 	[JITTER_OPT_RETRY_DELAY] = {
 		.key = "retry_delay",
-		.int_val = -1,
+		.int_val = 1,
 	},
 	[JITTER_OPT_USE_AES] = {
 		.key = "use_aes",
-		.int_val = 1,
+		.int_val = 0,
 	},
 	{
 		.key = NULL,
