--- rng-tools-6.5/jitterentropy-library/Makefile.orig	2020-12-05 15:54:14.538211185 -0600
+++ rng-tools-6.5/jitterentropy-library/Makefile	2020-12-05 15:55:33.066860296 -0600
@@ -4,6 +4,7 @@
 #Hardening
 CFLAGS ?= -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -fstack-protector-strong -fwrapv --param ssp-buffer-size=4 -fvisibility=hidden -fPIE -Wcast-align -Wmissing-field-initializers -Wshadow -Wswitch-enum
 CFLAGS +=-Wextra -Wall -pedantic -fPIC -O2 -fwrapv -Wconversion
+CFLAGS +=-fstack-protector-strong --param ssp-buffer-size=4
 LDFLAGS +=-Wl,-z,relro,-z,now
 
 # Change as necessary
