--- rng-tools-6.3.1/configure.ac.orig	2018-08-26 10:24:51.653374700 -0500
+++ rng-tools-6.3.1/configure.ac	2018-08-26 10:27:32.498703204 -0500
@@ -48,8 +48,8 @@
 AC_PROG_GCC_TRADITIONAL
 AC_CANONICAL_HOST
 
-AM_CONDITIONAL([RDRAND], [test $host_cpu = x86_64 -o $host_cpu = i686])
-AS_IF([test $host_cpu = x86_64 -o $host_cpu = i686], [AC_DEFINE([HAVE_RDRAND],1,[Enable RDRAND])],[])
+AM_CONDITIONAL([RDRAND], false)
+AS_IF(false, [AC_DEFINE([HAVE_RDRAND],1,[Enable RDRAND])],[])
 
 AM_CONDITIONAL([DARN], [test $host_cpu = powerpc64le])
 AS_IF([test $host_cpu = powerpc64le], [AC_DEFINE([HAVE_DARN],1,[Enable DARN])],[])
