diff -Naur a/scripts/build/companion_libs/140-mpc.sh b/scripts/build/companion_libs/140-mpc.sh
--- a/scripts/build/companion_libs/140-mpc.sh	2018-04-15 22:04:28.750333055 -0400
+++ b/scripts/build/companion_libs/140-mpc.sh	2018-04-15 22:06:06.300662257 -0400
@@ -13,7 +13,7 @@
 # Download MPC
 do_mpc_get() {
     CT_GetFile "mpc-${CT_MPC_VERSION}" .tar.gz      \
-        http://www.multiprecision.org/mpc/download
+        https://ftp.gnu.org/gnu/mpc
 }
 
 # Extract MPC
