--- neon-0.32.1/Makefile.in.orig	2021-10-06 10:57:58.322850997 -0500
+++ neon-0.32.1/Makefile.in	2021-10-06 10:59:10.560854337 -0500
@@ -128,13 +128,13 @@
 
 install-docs: install-man install-html
 
-install-html: docs-html
+install-html:
 	$(INSTALL) -d $(DESTDIR)$(docdir)/html
 	for d in doc/html/*.html; do \
 		$(INSTALL_DATA) $$d $(DESTDIR)$(docdir)/html; \
 	done
 
-install-man: docs-man
+install-man:
 	$(INSTALL) -d $(DESTDIR)$(man3dir)
 	$(INSTALL) -d $(DESTDIR)$(man1dir)
 	for m in doc/man/*.3; do \
