Index: /pjproject/trunk/pjsip/src/pjsua2/endpoint.cpp
===================================================================
--- a/pjsip/src/pjsua2/endpoint.cpp	(revision 5890)
+++ b/pjsip/src/pjsua2/endpoint.cpp	(revision 5891)
@@ -2146,7 +2146,8 @@
 StringVector Endpoint::srtpCryptoEnum() throw(Error)
 {
+    StringVector result;
+#if defined(PJMEDIA_HAS_SRTP) && (PJMEDIA_HAS_SRTP != 0)
     unsigned cnt = PJMEDIA_SRTP_MAX_CRYPTOS;
     pjmedia_srtp_crypto cryptos[PJMEDIA_SRTP_MAX_CRYPTOS];
-    StringVector result;
 
     PJSUA2_CHECK_EXPR(pjmedia_srtp_enum_crypto(&cnt, cryptos));
@@ -2154,4 +2155,5 @@
     for (unsigned i = 0; i < cnt; ++i)
 	result.push_back(pj2Str(cryptos[i].name));
+#endif
 
     return result;
