--- mISDN/config/Makefile.orig	2007-05-29 14:23:39.000000000 +0200
+++ mISDN/config/Makefile	2007-05-29 14:24:13.000000000 +0200
@@ -7,5 +7,5 @@
 	for file in $(shell echo *.xsl); do install -D -m644 $${file} $(INSTALL_PREFIX)/usr/lib/mISDN/$${file}; done
 	if [ -d $(INSTALL_PREFIX)/etc/init.d ]; then \
 		if [ -e $(INSTALL_PREFIX)/etc/init.d/mISDN ]; then rm -rf $(INSTALL_PREFIX)/etc/init.d/mISDN; fi; \
-		ln -s $(INSTALL_PREFIX)/usr/sbin/mISDN $(INSTALL_PREFIX)/etc/init.d/mISDN; \
+		ln -s ../../usr/sbin/mISDN $(INSTALL_PREFIX)/etc/init.d/mISDN; \
 	fi
