diff -urN sipsak-0.9.6.orig/transport.c sipsak-0.9.6/transport.c
--- sipsak-0.9.6.orig/transport.c	2006-01-28 16:11:34.000000000 -0500
+++ sipsak-0.9.6/transport.c	2010-10-26 18:38:51.000000000 -0400
@@ -286,7 +286,7 @@
 			}
 		}
 		senddiff = deltaT(&(srt->starttime), &(srt->recvtime));
-		if (senddiff > (float)64 * (float)SIP_T1) {
+		if (senddiff > inv_final) {
 			if (timing == 0) {
 				if (verbose>0)
 					printf("*** giving up, no final response after %.3f ms\n", senddiff);
