--- lighttpd-1.4.28.orig/src/request.c	2010-08-17 05:04:38.000000000 -0400
+++ lighttpd-1.4.28/src/request.c	2011-02-24 13:11:20.000000000 -0500
@@ -176,12 +176,12 @@
 					label_len = 0;
 					level++;
 				} else if (i == 0) {
-					if (!light_isalnum(c)) {
+					if (!light_isalnum(c) && c != '_') {
 						return -1;
 					}
 					label_len++;
 				} else {
-					if (c != '-' && !light_isalnum(c)) {
+					if (c != '-' && c != '_' &&!light_isalnum(c)) {
 						return -1;
 					}
 					label_len++;
