--- arp-scan-1.9/arp-scan.c.orig	2016-02-12 16:06:08.000000000 -0600
+++ arp-scan-1.9/arp-scan.c	2016-02-12 16:48:27.000000000 -0600
@@ -342,32 +342,21 @@
  * Create MAC/Vendor hash table if quiet is not in effect.
  */
    if (!quiet_flag) {
-      char *fn;
+      const char *list = "0123456789ABCDEF";
+      const char *cp;
+      char fn[32];
       int count;
 
       if ((hcreate(HASH_TABLE_SIZE)) == 0)
          err_sys("hcreate");
 
-      fn = get_mac_vendor_filename(ouifilename, DATADIR, OUIFILENAME);
-      count = add_mac_vendor(fn);
-      if (verbose > 1 && count > 0)
-         warn_msg("DEBUG: Loaded %d IEEE OUI/Vendor entries from %s.",
-                  count, fn);
-      free(fn);
-
-      fn = get_mac_vendor_filename(iabfilename, DATADIR, IABFILENAME);
-      count = add_mac_vendor(fn);
-      if (verbose > 1 && count > 0)
-         warn_msg("DEBUG: Loaded %d IEEE IAB/Vendor entries from %s.",
-                  count, fn);
-      free(fn);
-
-      fn = get_mac_vendor_filename(macfilename, DATADIR, MACFILENAME);
-      count = add_mac_vendor(fn);
-      if (verbose > 1 && count > 0)
-         warn_msg("DEBUG: Loaded %d MAC/Vendor entries from %s.",
-                  count, fn);
-      free(fn);
+      for (cp = list; *cp != '\0'; cp++) {
+         snprintf(fn, 32, "/usr/share/oui-db/xxxxx%c", *cp);
+         count = add_mac_vendor(fn);
+         if (verbose > 1 && count > 0)
+            warn_msg("DEBUG: Loaded %d IEEE OUI/Vendor entries from %s.",
+                     count, fn);
+      }
    }
 /*
  *      Populate the list from the specified file if --file was specified, or
@@ -2434,7 +2423,7 @@
 add_mac_vendor(const char *map_filename) {
    static int first_call=1;
    FILE *fp;	/* MAC/Vendor file handle */
-   static const char *oui_pat_str = "([^\t]+)\t[\t ]*([^\t\r\n]+)";
+   static const char *oui_pat_str = "([^~]+)~[\t ]*([^\t\r\n]+)";
    static regex_t oui_pat;
    regmatch_t pmatch[3];
    size_t key_len;
