--- nut-2.7.1/clients/upsc.c.orig	2014-01-05 22:44:13.000000000 -0600
+++ nut-2.7.1/clients/upsc.c	2014-01-05 22:55:08.000000000 -0600
@@ -27,6 +27,8 @@
 
 #include "upsclient.h"
 
+#define TIMEOUT		5
+
 static char		*upsname = NULL, *hostname = NULL;
 static UPSCONN_t	*ups = NULL;
 
@@ -206,6 +208,12 @@
 	free(ups);
 }
 
+static void noresp(int ign)
+{
+	printf("Error: No response.\n");
+	exit(EXIT_FAILURE);
+}
+
 int main(int argc, char **argv)
 {
 	int	i, port;
@@ -253,9 +261,12 @@
 
 	ups = xmalloc(sizeof(*ups));
 
+	signal(SIGALRM, noresp);
+	alarm(TIMEOUT);
 	if (upscli_connect(ups, hostname, port, UPSCLI_CONN_TRYSSL) < 0) {
 		fatalx(EXIT_FAILURE, "Error: %s", upscli_strerror(ups));
 	}
+	alarm(0);
 
 	if (varlist) {
 		list_upses(verbose);
