--- openssl-1.0.2j/Makefile.org.orig	2016-11-03 11:36:09.000000000 -0500
+++ openssl-1.0.2j/Makefile.org	2016-11-03 11:36:17.000000000 -0500
@@ -540,7 +540,7 @@
 	@$(MAKE) SDIRS='$(SDIRS)' clean
 	@$(MAKE) TAR='$(TAR)' TARFLAGS='$(TARFLAGS)' $(DISTTARVARS) tar
 
-install: all install_docs install_sw
+install: all install_sw
 
 install_sw:
 	@$(PERL) $(TOP)/util/mkdir-p.pl $(INSTALL_PREFIX)$(INSTALLTOP)/bin \
