From 50a7a07553c209765a645886ddc27c054d6c8641 Mon Sep 17 00:00:00 2001
From: Dirk Schneider <22996350+libertyx82@users.noreply.github.com>
Date: Thu, 9 Apr 2020 15:40:04 +0200
Subject: [PATCH] GUA only on IPv6

---
 ddclient | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ddclient b/ddclient
index 08dbdfd..15d0a9f 100755
--- a/ddclient
+++ b/ddclient
@@ -2289,7 +2289,7 @@ sub get_ipv6 {
     } elsif ($usev6 eq 'if') {
         $skip = opt('if-skip', $h) || '';
         $reply = `ip -6 -o addr show dev $arg scope global 2>/dev/null`;
-        if ($reply =~ /^.*? ([a-f0-9]{1,4}:[a-f0-9:]+:[a-f0-9]{1,4})\/.*/is) {
+        if ($reply =~ /^.*? ([23][a-f0-9]{3}:[a-f0-9:]+:[a-f0-9]{1,4})\/.*/is) {
             $reply = $1;
         } else {
             $reply = '';
