diff --git a/ddclient b/ddclient
index 2b7599a..7e1eb89 100755
--- a/ddclient
+++ b/ddclient
@@ -4580,7 +4580,7 @@ sub nic_he_update {
        'interval' => 'The update interval has been exceeded, wait and try again later',
        'noipv4'   => 'No DNS "A" record exists, add one and make it dynamic',
        'noipv6'   => 'No DNS "AAAA" record exists, add one and make it dynamic',
-       'nochg'    => 'No update required; unnecessary attempts to change to the current address are considered abusive',
+       'nochg'    => 'Unnecessary update attempts are considered abusive',
     );
 
     ## update each set of hosts that had similar configurations
@@ -4624,7 +4624,7 @@ sub nic_he_update {
 
                     } elsif (exists $errors{$status}) {
                         if (($status eq 'nochg') || ($status eq 'interval')) {
-                            warning("updating %s: %s: %s", $host, $status, $errors{$status});
+                            warning("updating %s: No DNS 'A' record update required; %s", $host, $errors{$status});
                             $config{$host}{'ip'}     = $ip;
                             $config{$host}{'mtime'}  = $now;
                             $config{$host}{'status'} = 'good';
@@ -4669,7 +4669,7 @@ sub nic_he_update {
 
                     } elsif (exists $errors{$status}) {
                         if (($status eq 'nochg') || ($status eq 'interval')) {
-                            warning("updating %s: %s: %s", $host, $status, $errors{$status});
+                            warning("updating %s: No DNS 'AAAA' record update required; %s", $host, $errors{$status});
                             $config{$host}{'ipv6'}   = $ipv6;
                             $config{$host}{'mtime'}  = $now;
                             $config{$host}{'status'} = 'good';
