--- a/ddclient	2017-01-20 09:04:33.026792149 -0600
+++ b/ddclient	2017-01-20 09:06:01.478773577 -0600
@@ -1878,13 +1878,13 @@
 ## load_json_support
 ######################################################################
 sub load_json_support {
-	my $json_loaded = eval {require JSON::Any};
+	my $json_loaded = eval {require JSON::PP};
 	unless ($json_loaded) {
 		fatal(<<"EOM");
-Error loading the Perl module JSON::Any needed for cloudflare update.
+Error loading the Perl module JSON::PP needed for cloudflare update.
 EOM
 	}
-	import JSON::Any;
+	import JSON::PP (qw/decode_json/);
 }
 ######################################################################
 ## geturl
@@ -4098,9 +4098,9 @@
 
 			# Strip header
 			$reply =~ s/^.*?\n\n//s;
-			my $response = JSON::Any->jsonToObj($reply);
-			if ($response->{result} eq 'error') {
-				failed ("%s", $response->{msg});
+			my $response = eval {decode_json($reply)};
+			if (!defined $response || !defined $response->{result}) {
+				failed ("invalid json or result.");
 				next; 
 			}
 
@@ -4125,9 +4125,9 @@
 
 			# Strip header
 			$reply =~ s/^.*?\n\n//s;
-			$response = JSON::Any->jsonToObj($reply);
-			if ($response->{result} eq 'error') {
-				failed ("%s", $response->{msg});
+			$response = eval {decode_json($reply)};
+			if (!defined $response || !defined $response->{result}) {
+				failed ("invalid json or result.");
 				next; 
 			}
 
@@ -4151,9 +4151,9 @@
 
 			# Strip header
 			$reply =~ s/^.*?\n\n//s;
-			$response = JSON::Any->jsonToObj($reply);
-			if ($response->{result} eq 'error') {
-				failed ("%s", $response->{msg});	
+			$response = eval {decode_json($reply)};
+			if (!defined $response || !defined $response->{result}) {
+				failed ("invalid json or result.");
 			} else {
 				success ("%s -- Updated Successfully to %s", $domain, $ip);
 
