--- dahdi-linux-2.6.1/drivers/dahdi/wcfxo.c.orig	2012-10-13 09:35:10.000000000 -0500
+++ dahdi-linux-2.6.1/drivers/dahdi/wcfxo.c	2012-10-13 09:35:18.000000000 -0500
@@ -240,6 +240,13 @@
 	{ "CTR21", 0, 0, 3, 0, 0, 3, 0 },	/* Austria, Belgium, Denmark, Finland, France, Germany, 
 										   Greece, Iceland, Ireland, Italy, Luxembourg, Netherlands,
 										   Norway, Portugal, Spain, Sweden, Switzerland, and UK */
+	{ "TBR21", 0, 1, 3, 0, 0, 3, 0 },	/* CTR21 with complex impedance */
+	{ "JATE", 0, 0, 1, 0, 0, 0, 0 },	/* Brazil, China, Egypt, Japan, Jordan, Kazakhstan, Malaysia,
+										   Oman, Pakistan, Philipines, Russia, Syria, Taiwan, Thailand */
+	{ "AUST", 1, 1, 1, 0, 0, 0, 0 },	/* Australia */
+	{ "NZWL", 0, 1, 2, 0, 0, 0, 0 },	/* New Zealand */
+	{ "PLSV", 0, 0, 2, 1, 1, 0, 0 },	/* Poland, Slovenia */
+	{ "SAFR", 1, 0, 2, 1, 0, 0, 0 },	/* South Africa */
 };
 
 static inline void wcfxo_transmitprep(struct wcfxo *wc, unsigned char ints)
