--- dahdi-linux-2.6.1/drivers/dahdi/Kbuild.orig	2012-05-22 16:22:09.000000000 -0500
+++ dahdi-linux-2.6.1/drivers/dahdi/Kbuild	2012-05-22 16:23:11.000000000 -0500
@@ -37,6 +37,7 @@
 # drivers/staging/echo . The DAHDI OSLEC echo canceller will be built as
 # well:
 ifneq (,$(wildcard $(src)/../staging/echo/echo.c))
+obj-$(DAHDI_BUILD_ALL) += ../staging/echo/echo.o
 obj-m += dahdi_echocan_oslec.o
 obj-m += ../staging/echo/
 endif
