--- asterisk-16.4.0/channels/chan_dahdi.c.orig	2019-05-30 12:26:01.000000000 -0500
+++ asterisk-16.4.0/channels/chan_dahdi.c	2019-06-20 08:54:46.129608872 -0500
@@ -1766,8 +1766,8 @@
 
 	ast_assert(dahdi_channel != NULL);
 
-	blob = ast_json_pack("{s: i, s: i, s: s}",
-		"group", group,
+	blob = ast_json_pack("{s: I, s: i, s: s}",
+		"group", (ast_json_int_t)group,
 		"span", span,
 		"channel", dahdi_channel);
 	if (!blob) {
