--- asterisk-20/channels/chan_sip.c.orig	2015-04-23 10:22:04.000000000 -0500
+++ asterisk-20/channels/chan_sip.c	2015-04-23 10:22:40.000000000 -0500
@@ -17707,7 +17707,7 @@
 	}
 
 	if (!force) {
-		ast_verb(2, "Extension Changed %s[%s] new state %s for Notify User %s %s\n", exten, context, ast_extension_state2str(data->state), p->username,
+		ast_verb(5, "Extension Changed %s[%s] new state %s for Notify User %s %s\n", exten, context, ast_extension_state2str(data->state), p->username,
 				ast_test_flag(&p->flags[1], SIP_PAGE2_STATECHANGEQUEUE) ? "(queued)" : "");
 	}
 
