--- asterisk-11/Makefile.orig	2016-10-09 17:19:53.204176413 -0500
+++ asterisk-11/Makefile	2016-10-09 17:21:11.339716525 -0500
@@ -98,8 +98,8 @@
 export LDCONFIG_FLAGS
 export PYTHON
 
-# makeopts is required unless the goal is clean or distclean
-ifeq ($(findstring clean,$(MAKECMDGOALS)),)
+# makeopts is required unless the goal is distclean
+ifeq ($(findstring distclean,$(MAKECMDGOALS)),)
 include makeopts
 endif
 
