--- asterisk-20/main/loader.c.orig	2021-05-12 13:29:13.282834486 -0500
+++ asterisk-20/main/loader.c	2021-05-12 13:31:30.082072430 -0500
@@ -2663,13 +2663,13 @@
 		}
 
 		if (ast_str_strlen(warning_msg)) {
-			ast_log(LOG_WARNING, "%s\n", ast_str_buffer(warning_msg));
+			ast_verb(5, "%s\n", ast_str_buffer(warning_msg));
 		}
 
 		ast_free(mod_name);
 #else
 		if (cur->info->support_level == AST_MODULE_SUPPORT_DEPRECATED) {
-			ast_log(LOG_WARNING, "The deprecated module '%s' has been loaded and is running, it may be removed in a future version\n", cur->resource);
+			ast_verb(5, "The deprecated module '%s' has been loaded and is running, it may be removed in a future version\n", cur->resource);
 		}
 #endif
 	}
--- asterisk-20/channels/chan_sip.c.orig	2022-01-10 12:39:58.043215949 +0100
+++ asterisk-20/channels/chan_sip.c	2022-01-10 12:42:31.761072628 +0100
@@ -35512,10 +35512,10 @@
 
 static void deprecation_notice(void)
 {
-	ast_log(LOG_WARNING, "chan_sip has no official maintainer and is deprecated.  Migration to\n");
-	ast_log(LOG_WARNING, "chan_pjsip is recommended.  See guides at the Asterisk Wiki:\n");
-	ast_log(LOG_WARNING, "https://wiki.asterisk.org/wiki/display/AST/Migrating+from+chan_sip+to+res_pjsip\n");
-	ast_log(LOG_WARNING, "https://wiki.asterisk.org/wiki/display/AST/Configuring+res_pjsip\n");
+	ast_verb(5, "chan_sip has no official maintainer and is deprecated.  Migration to\n");
+	ast_verb(5, "chan_pjsip is recommended.  See guides at the Asterisk Wiki:\n");
+	ast_verb(5, "https://wiki.asterisk.org/wiki/display/AST/Migrating+from+chan_sip+to+res_pjsip\n");
+	ast_verb(5, "https://wiki.asterisk.org/wiki/display/AST/Configuring+res_pjsip\n");
 }
 
 /*! \brief Event callback which indicates we're fully booted */

