--- asterisk-18/main/cdr.c.orig	2019-04-24 17:19:32.560568313 -0500
+++ asterisk-18/main/cdr.c	2019-04-24 17:35:58.803723662 -0500
@@ -3546,6 +3546,13 @@
 			continue;
 		}
 
+		/* When a Dial() has a B() option 'Return(IgnoreNoAnswerCDR)' ignore unanswered events */
+		if (cdr->disposition < AST_CDR_ANSWERED &&
+				(!strcasecmp(cdr->lastapp, "return") && !strcasecmp(cdr->lastdata, "IgnoreNoAnswerCDR"))) {
+			ast_debug(1, "Skipping CDR for IgnoreNoAnswerCDR to %s since we weren't answered\n", cdr->dstchannel);
+			continue;
+		}
+
 		/* Modify CDR's */
 		AST_RWLIST_RDLOCK(&mo_list);
 		AST_RWLIST_TRAVERSE(&mo_list, i, list) {
