--- arno-iptables-firewall_2.0.1d/share/arno-iptables-firewall/environment.orig
+++ arno-iptables-firewall_2.0.1d/share/arno-iptables-firewall/environment
@@ -4,10 +4,8 @@
 #
 #                           ~ In memory of my dear father ~
 #
-# (C) Copyright 2001-2012 by Arno van Amersfoort
-# Co-authors            : Lonnie Abelbeck & Philip Prindeville
+# (C) Copyright 2001-2014 by Arno van Amersfoort & Lonnie Abelbeck
 # Homepage              : http://rocky.eld.leidenuniv.nl/
-# Freshmeat homepage    : http://freshmeat.net/projects/iptables-firewall/?topic_id=151
 # Email                 : a r n o v a AT r o c k y DOT e l d DOT l e i d e n u n i v DOT n l
 #                         (note: you must remove all spaces and substitute the @ and the .
 #                         at the proper locations!)
@@ -46,16 +44,6 @@
 
 ################################# Functions ####################################
 
-trace()
-{
-  if [ -n "$TRACEFILE" ]; then
-   ((PS4='' ; set -x ; : "$@" >/dev/null) 2>&1 ) | sed 's/^: //' >> $TRACEFILE
-  else
-    "$@"
-  fi
-}
-
-
 # Find command path with '/hint/path/command' as the argument
 find_command()
 {
@@ -303,7 +291,7 @@
 {
   local result retval IFS=' '
 
-  result=`trace $IP4TABLES "$@" 2>&1`
+  result=`$IP4TABLES "$@" 2>&1`
   retval=$?
 
   if [ $retval -ne 0 ]; then
@@ -330,7 +318,7 @@
 {
   local result retval IFS=' '
 
-  result=`trace $IP6TABLES "$@" 2>&1`
+  result=`$IP6TABLES "$@" 2>&1`
   retval=$?
 
   if [ $retval -ne 0 ]; then
@@ -429,7 +417,7 @@
 {
   local IFS=' '
 
-  trace $IP4TABLES "$@" >/dev/null 2>&1
+  $IP4TABLES "$@" >/dev/null 2>&1
 }
 
 
@@ -437,7 +425,7 @@
 {
   local IFS=' '
 
-  trace $IP6TABLES "$@" >/dev/null 2>&1
+  $IP6TABLES "$@" >/dev/null 2>&1
 }
 
 
@@ -452,7 +440,7 @@
     # Make sure environment variable is not set
     MODPROBE_OPTIONS=""
 
-    result=`trace $MODPROBE $@ 2>&1`
+    result=`$MODPROBE $@ 2>&1`
     retval=$?
 
     if [ $retval -ne 0 ]; then
@@ -514,7 +502,7 @@
         modprobe_commandline="$modprobe_commandline $module"
 
         IFS=' '
-        result=`trace $modprobe_commandline 2>&1`
+        result=`$modprobe_commandline 2>&1`
         retval=$?
 
         if [ $retval -ne 0 ]; then
@@ -554,7 +542,7 @@
 {
   local result retval IFS=' '
 
-  result=`trace $SYSCTL "$@" 2>&1`
+  result=`$SYSCTL "$@" 2>&1`
   retval=$?
 
   if [ $retval -ne 0 ]; then
@@ -594,7 +582,7 @@
       fi
       sysctl_commandline="$sysctl_commandline $variable"
 
-      result=`trace $sysctl_commandline 2>&1`
+      result=`$sysctl_commandline 2>&1`
       retval=$?
 
       if [ $retval -eq 0 ]; then
@@ -645,7 +633,7 @@
 ###################
 tc()
 {
-  trace $TC "$@"
+  $TC "$@"
 }
 
 
@@ -653,7 +641,7 @@
 ###################
 ip()
 {
-  trace $IP "$@"
+  $IP "$@"
 }
 
 
@@ -767,7 +755,7 @@
     case $if1 in
     *+)
       if1="${if1%+}"
-      if1="$($IP link | awk "\$2 ~ /${if1}[0-9]+:/ { print substr(\$2, 1, length(\$2)-1); }" | tr '\n' ' ')"
+      if1="$(ip link | awk "\$2 ~ /${if1}[0-9]+:/ { print substr(\$2, 1, length(\$2)-1); }" | tr '\n' ' ')"
       if [ -z "$if1" ]; then
         echo "wildcard: $if0 unmatched!" >&2
         continue
@@ -1654,14 +1642,6 @@
   IPV6_SUPPORT=0
 fi
 
-# check for tracing
-###################
-if [ "$TRACE" = "1" ]; then
-  TRACEFILE="/tmp/aif-trace.`date '+%Y%m%d-%H:%M:%S'`"
-  cp /dev/null $TRACEFILE
-  DISABLE_IPTABLES_BATCH=1
-fi
-
 # Default NAT_INTERNAL_NET to INTERNAL_NET, if not specified
 ############################################################
 if [ -z "$NAT_INTERNAL_NET" ]; then
@@ -1689,3 +1669,4 @@
     PLUGIN_CONF_PATH="/etc/arno-iptables-firewall/plugins"
   fi
 fi
+
