--- arno-iptables-firewall_2.0.1c/share/arno-iptables-firewall/environment.orig
+++ arno-iptables-firewall_2.0.1c/share/arno-iptables-firewall/environment
@@ -307,13 +307,19 @@
     # Show any (error) messages in red
     printf "\033[40m\033[1;31m${IP4TABLES} $*\nERROR ($retval): ${result}\033[0m\n" >&2
     if note_iptables_error "$@"; then
       RULE_WARNING=1
     fi
   elif [ -n "$result" ]; then
-    echo "${INDENT}$result"
+    # Filter successful, informational results
+    case "$result" in
+      'WARNING: '*' match is obsolete'*)
+        ;;
+      *) echo "${INDENT}$result"
+        ;;
+    esac
   fi
 
   return $retval
 }
 
 
@@ -328,13 +334,19 @@
     # Show any (error) messages in red
     printf "\033[40m\033[1;31m${IP6TABLES} $*\nERROR ($retval): ${result}\033[0m\n" >&2
     if note_iptables_error "$@"; then
       RULE_WARNING=1
     fi
   elif [ -n "$result" ]; then
-    echo "${INDENT}$result"
+    # Filter successful, informational results
+    case "$result" in
+      'WARNING: '*' match is obsolete'*)
+        ;;
+      *) echo "${INDENT}$result"
+        ;;
+    esac
   fi
 
   return $retval
 }
 
