diff --git a/bin/arno-iptables-firewall b/bin/arno-iptables-firewall
index 99b7c9a..cc97aea 100755
--- a/bin/arno-iptables-firewall
+++ b/bin/arno-iptables-firewall
@@ -1074,19 +1074,16 @@ setup_valid_chk_chain()
   ######################
   iptables -A VALID_CHK $NF_CONNTRACK_STATE INVALID -j POST_INPUT_DROP_CHAIN
 
-  # Fragmentation cannot happen with IPV6 (and maybe even not with iptables/ipv4?)
-  ## Log fragmented packets
-  #########################
-  if [ "$FRAG_LOG" = "1" ]; then
-    echo "Logging of fragmented packets enabled"
+  # Fragmented packets handling
+  # NOTE: Fragmentation cannot happen with IPv6 (and probably even not with iptables/IPv4)
+  ########################################################################################
+  if [ "$FRAG_DROP" != "0" ]; then
+    echo "Dropping and logging of IPv4 fragmented packets enabled"
     ip4tables -A VALID_CHK -f -m limit --limit 3/m --limit-burst 1 -j LOG --log-prefix "AIF:Fragment packet: "
+    ip4tables -A VALID_CHK -f -j DROP
   else
-    echo "Logging of IPv4 fragmented packets disabled"
+    echo "Dropping and logging of IPv4 fragmented packets disabled"
   fi
-
-  # Drop fragmented packets
-  #########################
-  ip4tables -A VALID_CHK -f -j DROP
 }
 
 
@@ -5016,12 +5013,14 @@ setup_firewall_rules()
   ######################
   plugins_start
 
-  # Drop outgoing fragmented packets (this should in principle never happen because of netfilter's
-  # packet defragmentation
-  ################################################################################################
-  ip4tables -A OUTPUT -f \
-    -m limit --limit 3/m -j LOG --log-level $LOGLEVEL --log-prefix "AIF:Fragment packet: "
-  ip4tables -A OUTPUT -f -j DROP
+  # Fragmented packets handling
+  # NOTE: Fragmentation cannot happen with IPv6 (and probably even not with iptables/IPv4)
+  ########################################################################################
+  if [ "$FRAG_DROP" != "0" ]; then
+    ip4tables -A OUTPUT -f \
+      -m limit --limit 3/m -j LOG --log-level $LOGLEVEL --log-prefix "AIF:Fragment packet: "
+    ip4tables -A OUTPUT -f -j DROP
+  fi
 
   # Setup helper chains for EXTERNAL input traffic:
   echo "Setting up external(INET) INPUT policy"
diff --git a/etc/arno-iptables-firewall/firewall.conf b/etc/arno-iptables-firewall/firewall.conf
index 1a52706..686808f 100644
--- a/etc/arno-iptables-firewall/firewall.conf
+++ b/etc/arno-iptables-firewall/firewall.conf
@@ -441,9 +441,10 @@ INVALID_ICMP_LOG=0
 # ------------------------------------------------------------------------------
 RESERVED_NET_LOG=0
 
-# Enable logging of fragmented packets.
+# Enable dropping (& logging) of IPv4 fragmented packets. Normally fragmented
+# packets are automatically defragmented by the kernel.
 # ------------------------------------------------------------------------------
-FRAG_LOG=1
+FRAG_DROP=1
 
 # Enable logging of denied local (OUTPUT) connections.
 # ------------------------------------------------------------------------------
