--- arno-iptables-firewall_2.0.1b/bin/arno-iptables-firewall.orig	(revision 638)
+++ arno-iptables-firewall_2.0.1b/bin/arno-iptables-firewall	(revision 639)
@@ -1307,12 +1307,13 @@
   # Allow to send ICMP packets?
   #############################
   if [ "$DMZ_OPEN_ICMP" != "0" ]; then
     echo " Allowing ICMP-requests(ping)"
     iptables -A DMZ_INPUT_CHAIN -p icmp --icmp-type echo-request -m limit --limit 20/second --limit-burst 100 -j ACCEPT
     if [ "$IPV6_SUPPORT" = "1" ]; then
+      unset IFS
       for icmpv6_type in $ICMPV6_SPECIAL_TYPES; do
         ip6tables -A DMZ_INPUT_CHAIN -p icmpv6 --icmpv6-type $icmpv6_type -m hl --hl-eq 255 -j ACCEPT
       done
     fi
   fi
 
