diff --git a/bin/arno-iptables-firewall b/bin/arno-iptables-firewall
index 043a010..99b7c9a 100755
--- a/bin/arno-iptables-firewall
+++ b/bin/arno-iptables-firewall
@@ -1019,24 +1019,6 @@ setup_valid_chk_chain()
   ################
   iptables -A VALID_CHK -p tcp --tcp-flags ALL NONE -j POST_INPUT_DROP_CHAIN
 
-  # Log packets with bad flags?
-  #############################
-  if [ "$BAD_FLAGS_LOG" != "0" ]; then
-    echo "Logging of packets with bad TCP-flags enabled"
-    iptables -A VALID_CHK -p tcp --tcp-option 64 \
-      -m limit --limit 3/m --limit-burst 1 -j LOG --log-level $LOGLEVEL --log-prefix "AIF:Bad TCP flag(64): "
-
-    iptables -A VALID_CHK -p tcp --tcp-option 128 \
-      -m limit --limit 3/m --limit-burst 1 -j LOG --log-level $LOGLEVEL --log-prefix "AIF:Bad TCP flag(128): "
-  else
-    echo "Logging of packets with bad TCP-flags disabled"
-  fi
-
-  # Drop packets with bad tcp flags
-  #################################
-  iptables -A VALID_CHK -p tcp --tcp-option 64 -j POST_INPUT_DROP_CHAIN
-  iptables -A VALID_CHK -p tcp --tcp-option 128 -j POST_INPUT_DROP_CHAIN
-
   # These packets are normally from "lost connection" and thus can generate false alarms
   # So we might want to ignore such packets
   ######################################################################################

diff --git a/etc/arno-iptables-firewall/firewall.conf b/etc/arno-iptables-firewall/firewall.conf
index f6be649..1a52706 100644
--- a/etc/arno-iptables-firewall/firewall.conf
+++ b/etc/arno-iptables-firewall/firewall.conf
@@ -419,10 +419,6 @@ SCAN_LOG=1
 # ------------------------------------------------------------------------------
 POSSIBLE_SCAN_LOG=1
 
-# Enable logging for TCP-packets with bad flags.
-# ------------------------------------------------------------------------------
-BAD_FLAGS_LOG=1
-
 # Enable logging of invalid TCP packets. Keep disabled (0) by default to reduce
 # INVALID packets being logged because of lost (legimate) connections. When
 # debugging any problems, you should enable it (temporarily)!
