diff --git a/bin/arno-iptables-firewall b/bin/arno-iptables-firewall
index 9ec86fc..043a010 100755
--- a/bin/arno-iptables-firewall
+++ b/bin/arno-iptables-firewall
@@ -830,11 +830,6 @@ init_firewall_chains()
   # Create several chains that we will use later on
   #################################################
   create_user_chains
-
-  # Reset the iptables counters
-  iptables -Z
-  iptables -t mangle -Z
-  ip4tables -t nat -Z
 }
 
 
