--- darkstat-3.0.718/dns.c.orig	2014-01-25 20:39:41.000000000 -0600
+++ darkstat-3.0.718/dns.c	2014-01-25 22:30:49.000000000 -0600
@@ -41,6 +41,7 @@
 #define PARENT 1
 static int dns_sock[2];
 static pid_t pid = -1;
+static int log_write_error = 1;
 
 struct dns_reply {
    struct addr addr;
@@ -142,9 +143,12 @@
    num_w = write(dns_sock[PARENT], ipaddr, sizeof(*ipaddr)); /* won't block */
    if (num_w == 0)
       warnx("dns_queue: write: ignoring end of file");
-   else if (num_w == -1)
-      warn("dns_queue: ignoring write error");
-   else if (num_w != sizeof(*ipaddr))
+   else if (num_w == -1) {
+      if (log_write_error) {  /* only log once as this can happen often */
+         warn("dns_queue: ignoring write error");
+         log_write_error = 0;
+      }
+   } else if (num_w != sizeof(*ipaddr))
       err(1, "dns_queue: wrote %zu instead of %zu", num_w, sizeof(*ipaddr));
 }
 
