--- darkstat-3.0.717/html.c.orig	2013-11-14 13:36:46.000000000 -0600
+++ darkstat-3.0.717/html.c	2013-11-14 13:39:51.000000000 -0600
@@ -53,12 +53,13 @@
          "<li class=\"label\">" PACKAGE_STRING "</li>"
          "<li><a href=\"%s/\">graphs</a></li>"
          "<li><a href=\"%s/hosts/\">hosts</a></li>"
-         "<li><a href=\"" PACKAGE_URL "\">homepage</a></li>"
+         "<li class=\"label\">%s</li>"
+         "<li class=\"label\">" PACKAGE_URL "</li>"
         "</ul>\n"
         "</div>\n"
         "<div class=\"content\">\n"
          "<h2 class=\"pageheader\">%s</h2>\n"
-        , root, root, title);
+        , root, root, title_interfaces, title);
 }
 
 void html_close(struct str *buf)
