--- wireguard-0.0.20190406/src/crypto/zinc/curve25519/curve25519.c.orig	2019-04-27 21:00:09.291211877 -0500
+++ wireguard-0.0.20190406/src/crypto/zinc/curve25519/curve25519.c	2019-04-27 21:02:13.569223713 -0500
@@ -20,7 +20,7 @@
 #include <linux/init.h>
 #include <crypto/algapi.h> // For crypto_memneq.
 
-#if defined(CONFIG_ZINC_ARCH_X86_64)
+#if 0 && defined(CONFIG_ZINC_ARCH_X86_64)
 #include "curve25519-x86_64-glue.c"
 #elif defined(CONFIG_ZINC_ARCH_ARM)
 #include "curve25519-arm-glue.c"
