From bca546c22a083b50aa24bfa3a78ac17e88d9292a Mon Sep 17 00:00:00 2001
From: "Jason A. Donenfeld" <Jason@zx2c4.com>
Date: Wed, 22 Nov 2017 20:21:48 +0100
Subject: compat: support timespec64 on old kernels

---
 src/compat/compat.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/compat/compat.h b/src/compat/compat.h
index 68590fa..4609417 100644
--- a/src/compat/compat.h
+++ b/src/compat/compat.h
@@ -523,6 +523,11 @@ struct _____dummy_container { char dev; };
 #define COMPAT_CANNOT_USE_AVX512
 #endif
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 17, 0)
+#define timespec64 timespec
+#define getnstimeofday64 getnstimeofday
+#endif
+
 /* https://lkml.org/lkml/2017/6/23/790 */
 #if IS_ENABLED(CONFIG_NF_CONNTRACK)
 #include <linux/ip.h>
-- 
cgit v1.1-33-gc3c0

