--- busybox-1.19.4/libbb/utmp.c.orig	2012-02-04 13:24:55.000000000 -0600
+++ busybox-1.19.4/libbb/utmp.c	2014-01-20 17:25:27.000000000 -0600
@@ -8,12 +8,6 @@
  */
 #include "libbb.h"
 
-static void touch(const char *filename)
-{
-	if (access(filename, R_OK | W_OK) == -1)
-		close(open(filename, O_WRONLY | O_CREAT, 0664));
-}
-
 void FAST_FUNC write_new_utmp(pid_t pid, int new_type, const char *tty_name, const char *username, const char *hostname)
 {
 	struct utmp utent;
@@ -45,12 +39,13 @@
 		tty_name += 3;
 	strncpy(id, tty_name, width);
 
-	touch(_PATH_UTMP);
-	//utmpname(_PATH_UTMP);
-	setutent();
-	/* Append new one (hopefully, unless we collide on ut_id) */
-	pututline(&utent);
-	endutent();
+	if (access(_PATH_UTMP, R_OK | W_OK) != -1) {
+		//utmpname(_PATH_UTMP);
+		setutent();
+		/* Append new one (hopefully, unless we collide on ut_id) */
+		pututline(&utent);
+		endutent();
+	}
 
 #if ENABLE_FEATURE_WTMP
 	/* "man utmp" says wtmp file should *not* be created automagically */
@@ -67,7 +62,9 @@
 	struct utmp utent;
 	struct utmp *utp;
 
-	touch(_PATH_UTMP);
+	if (access(_PATH_UTMP, R_OK | W_OK) == -1) {
+		return;
+	}
 	//utmpname(_PATH_UTMP);
 	setutent();
 
