--- mtr-0.86/configure.ac.orig	2014-12-28 15:26:09.000000000 -0600
+++ mtr-0.86/configure.ac	2014-12-28 15:35:37.000000000 -0600
@@ -34,6 +34,9 @@
 	AC_DEFINE(NO_CURSES, 1, Define if you don't have the curses libraries available.)
 	CURSES_OBJ=))))
 
+AC_CHECK_LIB(ncurses, use_default_colors, 
+  AC_DEFINE(HAVE_USE_DEFAULT_COLORS, 1, [Define this if your curses library has the use_default_colors() command.]))
+
 AC_CHECK_FUNCS(attron fcntl)
 
 AC_CHECK_LIB(m, floor, , AC_MSG_ERROR(No math library found))
--- mtr-0.86/curses.c.orig	2014-12-28 14:57:03.000000000 -0600
+++ mtr-0.86/curses.c	2014-12-28 14:58:19.000000000 -0600
@@ -701,6 +701,9 @@
   raw();
   noecho(); 
   start_color();
+#ifdef HAVE_USE_DEFAULT_COLORS
+  use_default_colors();
+#endif
   int i;
   for (i = 0; i < 8; i++)
       init_pair(i+1, i, 0);

From 7571201cf7a3394e0dcd2b037aba1836089cc084 Mon Sep 17 00:00:00 2001
From: Narthorn <narthorn@gmail.com>
Date: Mon, 12 Oct 2015 13:24:57 +0200
Subject: [PATCH] curses: Fix background transparency in terminal

Patch comes from, and closes traviscross/mtr#72.
---
 curses.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/curses.c b/curses.c
index 02b7937..f95f5d1 100644
--- a/curses.c
+++ b/curses.c
@@ -700,13 +700,15 @@ void mtr_curses_open(void)
   initscr();
   raw();
   noecho(); 
+  int bg_col = 0;
   start_color();
 #ifdef HAVE_USE_DEFAULT_COLORS
-  use_default_colors();
+  if (use_default_colors() == OK)
+    bg_col = -1;
 #endif
   int i;
   for (i = 0; i < 8; i++)
-      init_pair(i+1, i, 0);
+      init_pair(i+1, i, bg_col);
 
   mtr_curses_init();
   mtr_curses_redraw();
