--- mtr-0.86/configure.ac.orig	2014-12-28 15:26:09.000000000 -0600
+++ mtr-0.86/configure.ac	2014-12-28 15:35:37.000000000 -0600
@@ -34,6 +34,9 @@
 	AC_DEFINE(NO_CURSES, 1, Define if you don't have the curses libraries available.)
 	CURSES_OBJ=))))
 
+AC_CHECK_LIB(ncurses, use_default_colors, 
+  AC_DEFINE(HAVE_USE_DEFAULT_COLORS, 1, [Define this if your curses library has the use_default_colors() command.]))
+
 AC_CHECK_FUNCS(attron fcntl)
 
 AC_CHECK_LIB(m, floor, , AC_MSG_ERROR(No math library found))
--- mtr-0.86/curses.c.orig	2014-12-28 14:57:03.000000000 -0600
+++ mtr-0.86/curses.c	2014-12-28 14:58:19.000000000 -0600
@@ -701,6 +701,9 @@
   raw();
   noecho(); 
   start_color();
+#ifdef HAVE_USE_DEFAULT_COLORS
+  use_default_colors();
+#endif
   int i;
   for (i = 0; i < 8; i++)
       init_pair(i+1, i, 0);
