diff --git a/src/actions.c b/src/actions.c
index 1cbe903..515023b 100644
--- a/src/actions.c
+++ b/src/actions.c
@@ -896,6 +896,10 @@ int do_action(buffer *b, action a, int64_t c, char *p) {
 		p = str_dup(b->filename);
 
 	case SAVEAS_A:
+		if (b->opt.read_only && !request_response(b, info_msg[SAVE_READ_ONLY_DOCUMENT],false)) {
+			free(p);
+			return DOCUMENT_NOT_SAVED;
+		}
 		if (p || (q = p = request_file(b, "Filename", b->filename))) {
 			print_info(SAVING);
 
@@ -903,7 +907,10 @@ int do_action(buffer *b, action a, int64_t c, char *p) {
 				free(p);
 				return DOCUMENT_NOT_SAVED;
 			}
+			c = b->opt.read_only;
+			SET_USER_FLAG(b, 0, opt.read_only);
 			error = save_buffer_to_file(b, p);
+			SET_USER_FLAG(b, c, opt.read_only);
 
 			if (!print_error(error)) {
 				const bool load_syntax = b->filename == NULL || ! same_str(extension(p), extension(b->filename));
diff --git a/src/buffer.c b/src/buffer.c
index 6ea297a..890331a 100644
--- a/src/buffer.c
+++ b/src/buffer.c
@@ -1520,6 +1520,7 @@ int save_buffer_to_file(buffer *b, const char *name) {
 
 	assert_buffer(b);
 
+	if (b->opt.read_only) return DOCUMENT_IS_READ_ONLY;
 	if (name == NULL) name = b->filename;
 
 	if (!name) return ERROR;
diff --git a/src/errors.c b/src/errors.c
index 9053ce4..5626409 100644
--- a/src/errors.c
+++ b/src/errors.c
@@ -99,6 +99,7 @@ char *info_msg[INFO_COUNT] = {
 	"Saving...",
 	"Saved.",
 	"All modified documents saved.",
+	"Document is marked ReadOnly; save anyway?",
 	"SELECT cursor/Enter FILTER chars/BS MODE Ins/Del EDIT Tab FILENAME F1/Esc-Esc/Esc",
 	"Start of block marked. Move to the end of block and request actions.",
 	"Start of vertical block marked. Move to the end of block and request actions.",
diff --git a/src/errors.h b/src/errors.h
index b307a07..a217353 100644
--- a/src/errors.h
+++ b/src/errors.h
@@ -119,6 +119,7 @@ enum info {
 	SAVING,
 	SAVED,
 	MODIFIED_SAVED,
+	SAVE_READ_ONLY_DOCUMENT,
 	PRESSF1,
 	BLOCK_START_MARKED,
 	VERTICAL_BLOCK_START_MARKED,
