--- ngrep-1.48.1/configure.ac.orig	2025-11-04 14:01:09.362264296 -0600
+++ ngrep-1.48.1/configure.ac	2025-11-04 14:09:07.831492091 -0600
@@ -153,8 +153,11 @@
 
 AS_IF([test $use_pcre2 = yes], [
   USE_PCRE2="1"
-  EXTRA_INCLUDES="$EXTRA_INCLUDES $(pcre2-config --cflags-posix)"
-  EXTRA_LIBS="$EXTRA_LIBS $(pcre2-config --libs-posix)"
+  if test -z "$NGREP_PCRE2_CONFIG_SCRIPT"; then
+    NGREP_PCRE2_CONFIG_SCRIPT="pcre2-config"
+  fi
+  EXTRA_INCLUDES="$EXTRA_INCLUDES $($NGREP_PCRE2_CONFIG_SCRIPT --cflags-posix)"
+  EXTRA_LIBS="$EXTRA_LIBS $($NGREP_PCRE2_CONFIG_SCRIPT --libs-posix)"
 ], [
   USE_PCRE2="0"
 
