--- asterisk-1.8.23.0/res/res_jabber.c.orig	2013-08-26 21:01:37.000000000 -0500
+++ asterisk-1.8.23.0/res/res_jabber.c	2013-08-26 21:10:23.000000000 -0500
@@ -3507,7 +3507,7 @@
 	iks *orig_pubsub = iks_find(pak->x, "pubsub");
 	struct aji_client *client;
 	if (!orig_pubsub) {
-		ast_log(LOG_ERROR, "Error isn't a PubSub error, why are we here?\n");
+		ast_debug(1, "Error isn't a PubSub error, why are we here?\n");
 		return IKS_FILTER_EAT;
 	}
 	orig_request = iks_child(orig_pubsub);
