--- asterisk-13.1.0/configure.ac.orig	2015-01-05 15:32:02.000000000 -0600
+++ asterisk-13.1.0/configure.ac	2015-01-05 15:41:42.000000000 -0600
@@ -883,7 +883,10 @@
 		[sem_t sem; return sem_init(&sem, 0, 0);])],
 	AC_MSG_RESULT(yes)
 	AC_DEFINE([HAS_WORKING_SEMAPHORE], 1, [Define to 1 if anonymous semaphores work.]),
-	AC_MSG_RESULT(no)
+	AC_MSG_RESULT(no),
+	AC_MSG_RESULT(cross-compile)
+	AC_MSG_NOTICE([WARNING: result yes guessed because of cross compilation])
+	AC_DEFINE([HAS_WORKING_SEMAPHORE], 1, [Define to 1 if anonymous semaphores work.])
 )
 
 LIBS="$save_LIBS"
