--- asterisk-13.16.0/configure.ac.orig	2017-06-07 11:56:41.261126679 -0500
+++ asterisk-13.16.0/configure.ac	2017-06-07 12:01:48.732815227 -0500
@@ -1104,7 +1104,8 @@
       [return eventfd(0, EFD_NONBLOCK | EFD_SEMAPHORE) == -1;])],
   AC_MSG_RESULT(yes)
   AC_DEFINE([HAVE_EVENTFD], 1, [Define to 1 if your system supports eventfd and the EFD_NONBLOCK and EFD_SEMAPHORE flags.]),
-  AC_MSG_RESULT(no)
+  AC_MSG_RESULT(no),
+  AC_MSG_RESULT(cross-compile)
 )
 
 AST_GCC_ATTRIBUTE(pure)
