--- asterisk-1.8.25.0/main/asterisk.c.orig	2013/12/16 16:36:52	403913
+++ asterisk-1.8.25.0/main/asterisk.c	2014/01/03 22:06:02	404861
@@ -3251,7 +3251,9 @@
 			live_dangerously = ast_true(v->value);
 		}
 	}
-	pbx_live_dangerously(live_dangerously);
+	if (!ast_opt_remote) {
+		pbx_live_dangerously(live_dangerously);
+	}
 	for (v = ast_variable_browse(cfg, "compat"); v; v = v->next) {
 		float version;
 		if (sscanf(v->value, "%30f", &version) != 1) {
