--- asterisk-1.8.32.3/channels/chan_sip.c.orig	2015-04-23 10:15:19.000000000 -0500
+++ asterisk-1.8.32.3/channels/chan_sip.c	2015-04-23 10:15:40.000000000 -0500
@@ -15121,7 +15121,7 @@
 			ast_set_flag(&p->flags[1], SIP_PAGE2_STATECHANGEQUEUE);
 		}
 	}
-	ast_verb(2, "Extension Changed %s[%s] new state %s for Notify User %s %s\n", exten, context, ast_extension_state2str(state), p->username,
+	ast_verb(4, "Extension Changed %s[%s] new state %s for Notify User %s %s\n", exten, context, ast_extension_state2str(state), p->username,
 			ast_test_flag(&p->flags[1], SIP_PAGE2_STATECHANGEQUEUE) ? "(queued)" : "");
 
 	sip_pvt_unlock(p);
