--- asterisk-13.1.0/configure.ac.orig	2015-01-02 12:41:14.000000000 -0600
+++ asterisk-13.1.0/configure.ac	2015-01-02 12:43:39.000000000 -0600
@@ -877,14 +877,7 @@
 )
 
 # Some platforms define sem_init(), but only support sem_open(). joyous.
-AC_MSG_CHECKING(for working unnamed semaphores)
-AC_RUN_IFELSE(
-	[AC_LANG_PROGRAM([#include <semaphore.h>],
-		[sem_t sem; return sem_init(&sem, 0, 0);])],
-	AC_MSG_RESULT(yes)
-	AC_DEFINE([HAS_WORKING_SEMAPHORE], 1, [Define to 1 if anonymous semaphores work.]),
-	AC_MSG_RESULT(no)
-)
+AC_DEFINE([HAS_WORKING_SEMAPHORE], 1, [Define to 1 if anonymous semaphores work.])
 
 LIBS="$save_LIBS"
 CFLAGS="$save_CFLAGS"
