--- asterisk-11.2.0/res/pjproject/aconfigure	2013/01/04 22:18:11	378582
+++ asterisk-11.2.0/res/pjproject/aconfigure	2013/01/04 22:18:21	378581
@@ -2401,7 +2401,7 @@
   program_prefix=${target_alias}-
 ac_config_headers="$ac_config_headers pjlib/include/pj/compat/os_auto.h pjlib/include/pj/compat/m_auto.h pjmedia/include/pjmedia/config_auto.h pjmedia/include/pjmedia-codec/config_auto.h pjsip/include/pjsip/sip_autoconf.h"
 
-ac_config_files="$ac_config_files build.mak build/os-auto.mak pjlib/build/os-auto.mak pjlib-util/build/os-auto.mak pjmedia/build/os-auto.mak pjsip/build/os-auto.mak third_party/build/os-auto.mak third_party/build/portaudio/os-auto.mak"
+ac_config_files="$ac_config_files build.mak build/os-auto.mak build/cc-auto.mak pjlib/build/os-auto.mak pjlib-util/build/os-auto.mak pjmedia/build/os-auto.mak pjsip/build/os-auto.mak third_party/build/os-auto.mak third_party/build/portaudio/os-auto.mak"
 
 
 
@@ -7799,6 +7799,7 @@
     "pjsip/include/pjsip/sip_autoconf.h") CONFIG_HEADERS="$CONFIG_HEADERS pjsip/include/pjsip/sip_autoconf.h" ;;
     "build.mak") CONFIG_FILES="$CONFIG_FILES build.mak" ;;
     "build/os-auto.mak") CONFIG_FILES="$CONFIG_FILES build/os-auto.mak" ;;
+    "build/cc-auto.mak") CONFIG_FILES="$CONFIG_FILES build/cc-auto.mak" ;;
     "pjlib/build/os-auto.mak") CONFIG_FILES="$CONFIG_FILES pjlib/build/os-auto.mak" ;;
     "pjlib-util/build/os-auto.mak") CONFIG_FILES="$CONFIG_FILES pjlib-util/build/os-auto.mak" ;;
     "pjmedia/build/os-auto.mak") CONFIG_FILES="$CONFIG_FILES pjmedia/build/os-auto.mak" ;;

--- asterisk-11.2.0/res/pjproject/aconfigure.ac	2013/01/04 22:18:11	378582
+++ asterisk-11.2.0/res/pjproject/aconfigure.ac	2013/01/04 22:18:21	378581
@@ -13,6 +13,7 @@
 		 ])
 AC_CONFIG_FILES([build.mak 
 		 build/os-auto.mak 
+		 build/cc-auto.mak
 		 pjlib/build/os-auto.mak 
 		 pjlib-util/build/os-auto.mak 
 		 pjmedia/build/os-auto.mak


--- asterisk-11.2.0/res/pjproject/build/common.mak	2013/01/04 22:18:11	378582
+++ asterisk-11.2.0/res/pjproject/build/common.mak	2013/01/04 22:18:21	378581
@@ -17,6 +17,15 @@
 -include cc-$(CC_NAME).mak
 
 #
+# Include auto configured compiler specification.
+# This will override the compiler settings above.
+# Currently this is made OPTIONAL, to prevent people
+# from getting errors because they don't re-run ./configure
+# after downloading new PJSIP.
+#
+-include $(PJDIR)/build/cc-auto.mak
+
+#
 # Include global machine specific definitions
 #
 include $(PJDIR)/build/m-$(MACHINE_NAME).mak
