--- php-5.3.22/configure.orig	2013-03-17 14:56:49.000000000 -0500
+++ php-5.3.22/configure	2013-03-17 14:58:16.000000000 -0500
@@ -5454,7 +5454,7 @@
     GCC_MAJOR_VERSION=`$CC -dumpversion | /usr/bin/sed -nE '1s/([0-9]+)\.[0-9]+\..*/\1/;1p'`
     ;;
   *)
-    GCC_MAJOR_VERSION=`$CC --version | $SED -n '1s/[^0-9]*//;1s/\..*//;1p'`
+    GCC_MAJOR_VERSION=`$CC -dumpversion | cut -d'.' -f1`
     ;;
   esac
   if test $GCC_MAJOR_VERSION -ge 4; then
