--- whois-5.2.15/whois.c.orig	2017-03-02 07:46:01.316070356 -0600
+++ whois-5.2.15/whois.c	2017-03-02 07:47:25.420525150 -0600
@@ -1010,7 +1010,9 @@
     hints.ai_family = AF_UNSPEC;
     hints.ai_socktype = SOCK_STREAM;
     hints.ai_flags = AI_ADDRCONFIG;
+#ifdef HAVE_GAI_IDN
     hints.ai_flags |= AI_IDN;
+#endif
 
     if ((err = getaddrinfo(server, port ? port : "nicname", &hints, &res))
 	    != 0) {
