diff --git a/src/curses/ui_call_list.c b/src/curses/ui_call_list.c
index 9d854fc..043d2d1 100644
--- a/src/curses/ui_call_list.c
+++ b/src/curses/ui_call_list.c
@@ -933,7 +933,7 @@ call_list_help(ui_t *ui)
     mvwprintw(help_win, 17, 2, "F5/Ctrl-L   Clear call list (can not be undone!)");
     mvwprintw(help_win, 18, 2, "F6/R        Show selected call messages in raw mode");
     mvwprintw(help_win, 19, 2, "F7/F        Show filter options");
-    mvwprintw(help_win, 20, 2, "F8/c        Turn on/off window colours");
+    mvwprintw(help_win, 20, 2, "F8/o        Show Settings");
     mvwprintw(help_win, 21, 2, "F10/t       Select displayed columns");
     mvwprintw(help_win, 22, 2, "i/I         Set display filter to invite");
     mvwprintw(help_win, 23, 2, "p           Stop/Resume packet capture");
