--- msmtp-1.8.15/src/mtls-libtls.c.orig	2021-05-26 14:52:35.476470319 -0500
+++ msmtp-1.8.15/src/mtls-libtls.c	2021-05-26 14:53:36.387603968 -0500
@@ -320,6 +320,8 @@
     if (no_certcheck)
     {
         tls_config_insecure_noverifycert(config);
+        tls_config_insecure_noverifyname(config);
+        tls_config_insecure_noverifytime(config);
     }
     else if (sha256_fingerprint && !no_certcheck)
     {
