diff --git a/src/Makefile.am b/src/Makefile.am
index eac7942..1ce4a67 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -27,4 +27,9 @@ AM_CPPFLAGS = $(tls_CFLAGS) $(libgsasl_CFLAGS) $(libidn_CFLAGS) $(libsecret_CFLA
 
 msmtp_LDADD =  $(tls_LIBS) $(libgsasl_LIBS) $(libidn_LIBS) $(libsecret_LIBS) $(LIBINTL)
 
-DEFS += -DSYSCONFDIR=\"$(sysconfdir)\" -DLOCALEDIR=\"$(localedir)\"
+DEFS += -DSYSCONFDIR=\"$(sysconfdir)\" -DLOCALEDIR=\"$(localedir)\" -DBINDIR=\"$(bindir)\"
+
+
+bin_PROGRAMS += msmtpd
+msmtpd_SOURCES = msmtpd.c
+
