--- a/configure.ac	2017-03-31 09:15:57.593261737 -0500
+++ b/configure.ac	2017-03-31 09:17:32.969716649 -0500
@@ -57,9 +57,9 @@
     LIBS=""
     AC_SEARCH_LIBS([readline],[edit],[
       AC_DEFINE([HAVE_EDITLINE],1,Define to use BSD editline)
-      READLINE_LIBS="$LIBS -ltinfo"
+      READLINE_LIBS="$LIBS"
       enable_readline=no
-    ],[],[-ltinfo])
+    ])
     AS_UNSET(ac_cv_search_readline)
     LIBS=$sLIBS
   ])
