Index: sqlite-3090200/Makefile.am
==================================================================
--- sqlite-3090200/Makefile.am
+++ sqlite-3090200/Makefile.am
@@ -4,16 +4,18 @@
 lib_LTLIBRARIES = libsqlite3.la
 libsqlite3_la_SOURCES = sqlite3.c
 libsqlite3_la_LDFLAGS = -no-undefined -version-info 8:6:8
 
 bin_PROGRAMS = sqlite3
-sqlite3_SOURCES = shell.c sqlite3.c sqlite3.h
-sqlite3_LDADD = @READLINE_LIBS@
+sqlite3_SOURCES = shell.c sqlite3.h
+EXTRA_sqlite3_SOURCES = sqlite3.c
+sqlite3_LDADD = @EXTRA_SHELL_OBJ@ @READLINE_LIBS@
+sqlite3_DEPENDENCIES = @EXTRA_SHELL_OBJ@
 sqlite3_CFLAGS = $(AM_CFLAGS)
 
 include_HEADERS = sqlite3.h sqlite3ext.h
 
 EXTRA_DIST = sqlite3.1 tea
 pkgconfigdir = ${libdir}/pkgconfig
 pkgconfig_DATA = sqlite3.pc
 
 man_MANS = sqlite3.1

Index: sqlite-3090200/configure.ac
==================================================================
--- sqlite-3090200/configure.ac.orig	2015-11-02 12:44:06.000000000 -0600
+++ sqlite-3090200/configure.ac	2015-11-13 12:18:14.000000000 -0600
@@ -4,6 +4,8 @@
 #
 #   --enable-threadsafe
 #   --enable-readline
+#   --enable-editline
+#   --enable-static-shell
 #   --enable-dynamic-extensions
 #
 
@@ -103,6 +105,21 @@
 AC_SUBST(JSON1_FLAGS)
 #-----------------------------------------------------------------------
 
+#-----------------------------------------------------------------------
+#   --enable-static-shell
+#
+AC_ARG_ENABLE(static-shell, [AS_HELP_STRING(
+  [--enable-static-shell], 
+  [statically link libsqlite3 into shell tool [default=yes]])], 
+  [], [enable_static_shell=yes])
+if test x"$enable_static_shell" == "xyes"; then
+  EXTRA_SHELL_OBJ=sqlite3.$OBJEXT
+else
+  EXTRA_SHELL_OBJ=libsqlite3.la
+fi
+AC_SUBST(EXTRA_SHELL_OBJ)
+#-----------------------------------------------------------------------
+
 AC_CHECK_FUNCS(posix_fallocate)
 
 #-----------------------------------------------------------------------
