--- lldpd-0.9.9/src/daemon/lldpd.c.orig	2018-03-02 22:29:08.149158903 -0600
+++ lldpd-0.9.9/src/daemon/lldpd.c	2018-03-02 22:35:22.361022161 -0600
@@ -796,7 +796,6 @@
 lldpd_get_os_release() {
 	static char release[1024];
 	char line[1024];
-	char *key, *val;
 	char *ptr1 = release;
 
 	log_debug("localchassis", "grab OS release");
@@ -800,25 +800,15 @@
 	char *ptr1 = release;
 
 	log_debug("localchassis", "grab OS release");
-	FILE *fp = fopen("/etc/os-release", "r");
-	if (!fp) {
-		log_debug("localchassis", "could not open /etc/os-release");
-		fp = fopen("/usr/lib/os-release", "r");
-	}
+	FILE *fp = fopen("/etc/astlinux-release", "r");
 	if (!fp) {
 		log_info("localchassis",
-		    "could not open either /etc/os-release or /usr/lib/os-release");
+		    "could not open /etc/astlinux-release");
 		return NULL;
 	}
 
-	while ((fgets(line, sizeof(line), fp) != NULL)) {
-		key = strtok(line, "=");
-		val = strtok(NULL, "=");
-
-		if (strncmp(key, "PRETTY_NAME", sizeof(line)) == 0) {
-			strlcpy(release, val, sizeof(line));
-			break;
-		}
+	if ((fgets(line, sizeof(line), fp) != NULL)) {
+		strlcpy(release, line, sizeof(line));
 	}
 	fclose(fp);
 
