--- htop-3.0.4/linux/LibSensors.c.orig	2020-12-28 14:58:02.906861193 -0600
+++ htop-3.0.4/linux/LibSensors.c	2020-12-28 15:04:34.794087518 -0600
@@ -84,7 +84,7 @@
    for (const sensors_chip_name *chip = sym_sensors_get_detected_chips(NULL, &n); chip; chip = sym_sensors_get_detected_chips(NULL, &n)) {
       char buffer[32];
       sym_sensors_snprintf_chip_name(buffer, sizeof(buffer), chip);
-      if (!String_startsWith(buffer, "coretemp") && !String_startsWith(buffer, "cpu_thermal"))
+      if (!String_startsWith(buffer, "coretemp") && !String_startsWith(buffer, "cpu_thermal") && !String_startsWith(buffer, "k10temp"))
          continue;
 
       int m = 0;
@@ -97,7 +97,7 @@
             continue;
 
          unsigned int tempId;
-         if (String_startsWith(label, "Package ")) {
+         if (String_startsWith(label, "Package ") || String_startsWith(label, "CPU Temp")) {
             tempId = 0;
          } else if (String_startsWith(label, "temp")) {
             /* Raspberry Pi has only temp1 */
